core/classes/AuthFlags.class.php

```diff
@@ -69,6 +69,17 @@ class AuthFlags {
 	 */
 	private $credentials_page = null;
 
+	/**
+	 * The page to use for validating credentials.  This can be the default validator page
+	 * provided by MantisBT, an auth plugin provided page that validates the credentials 
+	 * the credentials againts an external source (and creates user if autoprovision is set)
+	 *
+	 * @see $login_page
+	 * @see $logout_page
+	 * @var string|null
+	 */
+	private $authenticator_page = null;
```

Comment:
Is there a reason why this shouldn't be transparent from core and implemented as part of `$credentials_page` and potentially other pages related to it as per plugin implementation?  For example, you can imagine the credentials page asking for password, then redirecting to authenticator, then logging in the user.

Reply:
This is separate by intention. In the updated readme of SampleAuth I described it in detail. Main reason is, that `$credentials_page`  is mainly for SSO IMHO (with separate password request page), while `$authenticator_page` replaced the standard login.php for access validation - but uses the Mantis username/password aquisition pages.